p2p/protocols/accounting_test.go

```diff
@@ -168,16 +173,25 @@ func TestBalance(t *testing.T) {
 }
 
 func checkAccountingTestCases(t *testing.T, cases []testCase, acc *Accounting, peer *Peer, balance *dummyBalance, send bool) {
+	t.Helper()
 	for _, c := range cases {
 		var err error
-		var expectedResult int64
+		var expectedResult, cost int64
 		//reset balance before every check
 		balance.amount = 0
 		if send {
-			err = acc.Send(peer, c.size, c.msg)
+			cost, err = acc.Validate(peer, c.size, c.msg, Sender)
```

Comment:
acc.Validate will always return `{0, nil}` because `c.msg` is not a `PricedMessage`. (see `p2p/protocols/accounting.go:143`)

Reply:
? You sure?

`c.msg` is from `c`, which is an iteration variable from `cases`, which is an array of `testCase`. These `testCases` are filled for example in `TestBalance` with instances, and every test case has a different instance of `msg`, in some cases it uses message types (e.g. `perBytesMsgSenderPays` or `perUnitMsgReceiverPays` which **do** implement `PricedMessage`, check L58-96

Reply:
You are right here, I did not see that you created those dummy types on top of the test file.